impl/src/main/java/org/ehcache/internal/store/heap/OnHeapStore.java

```diff
@@ -292,9 +329,13 @@ public void put(final K key, final V value) throws CacheAccessException {
 
       if (entryActuallyAdded.get()) {
         putObserver.end(StoreOperationOutcomes.PutOutcome.PUT);
-        enforceCapacity(1);
+        enforceCapacity(delta);
```

Comment:
That delta estimate isn't a bad idea, but it shouldn't be recalculated each time and you should loop around enforceCapacity() until enough bytes have been freed to make sure the store's consumption goes below the configured max bytes.